app/lib/money/types.ts

```diff
@@ -2,47 +2,60 @@ import type { Big } from 'big.js';
 
 export type NumberInput = number | string | Big;
 
-export type MoneyInput = {
+/** supported currencies */
+export type Currency = 'USD' | 'BTC';
+
+export type UsdUnit = 'usd' | 'cent';
+export type BtcUnit = 'btc' | 'sat' | 'msat';
+
+/** Unit to denominate the given currency */
+export type CurrencyUnit<T extends Currency> = T extends 'USD'
+  ? UsdUnit
+  : T extends 'BTC'
+    ? BtcUnit
+    : never;
+
+export type MoneyInput<T extends Currency = Currency> = {
```

Comment:
when would it be something that is not currency? perhaps we can lose the generic here?

Reply:
Right now, the `T` in `Money` is inferred from the `Currency` in `MoneyInput`. Wouldn't removing this make it so we can't do that anymore?

Reply:
yeah I was wondering for both. do you envison using money with something that is not of type currency?

I guess we could keep like this but we could lose a bit complexity maybe if we don't think it will be used. Approving and letting you decide